### DISCUSSION TITLE
Error due exit code 1 in bash command

### DISCUSSION THREAD

**Discussion by CGarces** [community/community#26662] [Actions]
<p>I need to compare some version numbers and I’m using <code>dpkg --compare-versions</code></p>
<pre><code class="lang-auto">run: |
        dpkg --compare-versions "${{matrix.kernel_version}}" "ge" "5.17"
        if [ $? -eq "0" ];
</code></pre>
<p>When the versions are greater of equal to 5.17 is working fine, but when is lower, it returns exit code 1 and the next line (that evaluate $?) is ignored.</p>
<p>What is the correct way to evaluate the exit code of one shell command inside “run:”?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>By default the shell in <code>run</code> steps runs with <code>set -e</code>, so it terminates as soon as any command returns an error (AKA a non-zero exit code).</p>
<aside class="quote no-group" data-username="cgarces" data-post="1" data-topic="250646">
<div class="title">
<div class="quote-controls"></div>
 cgarces:</div>
<blockquote>
<p>What is the correct way to evaluate the exit code of one shell command inside “run:”?</p>
</blockquote>
</aside>
<p>Put the command directly into the <code>if</code> condition instead of relying on <code>$?</code>, like so:</p>
<pre><code class="lang-yaml">run: |
  if dpkg --compare-versions "${{matrix.kernel_version}}" "ge" "5.17"; then
      # things to do if the dpkg call returns 0
  else
      # things to do otherwise
  fi
</code></pre>
<p>That way a possible non-zero exit code is handled by the <code>if</code> and doesn’t make the shell exit. :slightly_smiling_face:</p>

**Comment by CGarces** [comment]
<p>Thanks <a class="mention" href="/airtower-luna">@airtower-luna</a><br>
Your hint about “-e” was the key.<br>
I have solved the issue with<br>
<code>shell: bash {0}</code></p>
